x-pack/plugins/security_solution/public/timelines/components/timeline/header/save_timeline.tsx

```diff
@@ -0,0 +1,235 @@
+  ({ timelineId, toggleSaveTimeline, onSaveTimeline, updateTitle, updateDescription }) => {
+    const timeline = useShallowEqualSelector((state) =>
+      timelineSelectors.selectTimeline(state, timelineId)
+    );
+
+    const { description, isSaving, savedObjectId, title, timelineType } = timeline;
+
+    const handleClick = useCallback(() => {
+      onSaveTimeline({
+        ...timeline,
+        id: timelineId,
+      });
+    }, [onSaveTimeline, timeline, timelineId]);
+
+    const { getButton } = useCreateTimelineButton({ timelineId, timelineType });
+
+    const discardTimelineButton = useMemo(
+      () =>
+        getButton({
+          title:
+            timelineType === TimelineType.template
+              ? i18n.DISCARD_TIMELINE_TEMPLATE
+              : i18n.DISCARD_TIMELINE,
+          outline: true,
+          iconType: '',
+          fill: false,
+        }),
+      [getButton, timelineType]
+    );
+
+    const modalHeader =
+      savedObjectId == null
+        ? timelineType === TimelineType.template
+          ? i18n.SAVE_TIMELINE_TEMPLATE
+          : i18n.SAVE_TIMELINE
+        : timelineType === TimelineType.template
+        ? i18n.NAME_TIMELINE_TEMPLATE
+        : i18n.NAME_TIMELINE;
+
+    const saveButtonTitle =
+      savedObjectId == null
+        ? timelineType === TimelineType.template
+          ? i18n.SAVE_TIMELINE_TEMPLATE
+          : i18n.SAVE_TIMELINE
+        : i18n.SAVE;
+
+    return (
+      <>
+        {isSaving && (
+          <EuiProgress size="s" color="primary" position="absolute" data-test-subj="progress-bar" />
+        )}
+        <EuiModalHeader data-test-subj="modal-header">{modalHeader}</EuiModalHeader>
+
+        <Wrapper>
+          <EuiFlexItem grow={true}>
+            <EuiFormRow label={TIMELINE_TITLE}>
+              <Name
+                timelineId={timelineId}
+                timelineType={timelineType}
+                title={title}
+                updateTitle={updateTitle}
+                width="100%"
+                marginRight={10}
+                disableAutoSave={true}
+                data-test-subj="save-timeline-name"
+              />
+            </EuiFormRow>
+            <EuiSpacer />
+          </EuiFlexItem>
+          <EuiFlexItem grow={true}>
+            <EuiFormRow label={DESCRIPTION}>
+              <Description
+                description={description}
+                timelineId={timelineId}
+                updateDescription={updateDescription}
+                isTextArea={true}
+                disableAutoSave={true}
+                marginRight={0}
+                data-test-subj="save-timeline-description"
+              />
+            </EuiFormRow>
+            <EuiSpacer />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup justifyContent="flexEnd">
+              <EuiFlexItem grow={false} component="span">
+                {savedObjectId == null ? (
```

Comment:
I'm wondering if the behavior shown in the gif below, where the `Discard Timeline` button is displayed for a new, non-mutated timeline, is a product of Timeline being in the current "in between" state, where we're just starting to implement the new design, or potentially a bug:

![discard-timeline](https://user-images.githubusercontent.com/4459398/97199504-b037ee00-1775-11eb-977f-03b415e22104.gif)

I think the intent of the `Discard Timeline` button shown in the gif above is to address the case when a user has performed an action that will result in unsaved changes being discarded, because the timeline was never given a name. I'm not sure if the `Discard Timeline` button should appear in the scenario shown above, because we created a new timeline, with no mutations.

@angorayc would you be willing to confirm the expected behavior with @marrasherrier?

Reply:
@angorayc @andrew-goldstein I think it's best to keep the 'discard' option for consistency.

Reply:
@monina-n looping you in as well for transparency!

Reply:
From what we've talked about before, this is the same modal that will also appear when you click 'X' to close an unsaved timeline. I agree with @andrew-goldstein that having "Discard timeline" would be an unexpected option when you click the pencil icon to add a name/description?  If I click the pencil icon, I would expect the options to be "Save" and "Cancel" with cancel closing the modal instead of deleting the entire timeline. 

I think we can potentially go with @andrew-goldstein's suggestion of not having the "discard button" on the pencil and only when we're closing the timeline. Another possible solution is having two modals: (1) clicking the pencil icon and having the options be "Save Timeline" and "Cancel" and (2) clicking the 'X' to close a timeline and having the options be "Save Timeline" and "Discard Timeline" with the "discard" button being in red and maybe having a note that if you don't save the timeline, it will delete it.

@marrasherrier @andrew-goldstein @angorayc what do you guys think?

Reply:
@monina-n thanks for clarifying, that makes sense to me. @angorayc here is the separate interaction more clearly mocked out: https://www.figma.com/file/QVxq6M4TFDJlcgFHGe9wYT/Timeline-Evolution?node-id=1034%3A27036